src/preprocessing/passes/skeleton_preprocessing.cpp

```diff
@@ -0,0 +1,483 @@
+/*********************                                                        */
+/*! \file skeleton_preprocessing.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Aina Niemetz, Haoze Wu
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Boolean skeleton preprocessing pass.
+ **
+ ** Run cryptominisat and try to solve all assigned theory literals.
+ **/
+
+ #include "preprocessing/passes/skeleton_preprocessing.h"
+
+ #ifdef CVC4_USE_CRYPTOMINISAT
+ #include <cryptominisat5/cryptominisat.h>
+ #endif
+
+ #include "prop/registrar.h"
+ #include "prop/cnf_stream.h"
+ #include "prop/sat_solver_factory.h"
+
+ #include "proof/proof_manager.h"
+
+ #include "context/cdo.h"
+ #include "options/proof_options.h"
+ #include "smt/smt_statistics_registry.h"
+ #include "theory/theory_model.h"
+
+#include <vector>
+
+using namespace CVC4;
+using namespace CVC4::theory;
+
+namespace CVC4 {
+namespace preprocessing {
+namespace passes {
+
+
+/* -------------------------------------------------------------------------- */
+
+SkeletonPreprocessing::SkeletonPreprocessing(PreprocessingPassContext* preprocContext)
+    : PreprocessingPass(preprocContext, "non-clausal-simp-sat")
+{
+}
+
+PreprocessingPassResult SkeletonPreprocessing::applyInternal(
+    AssertionPipeline* assertionsToPreprocess)
+{
+
+  Assert(!options::unsatCores() && !options::fewerPreprocessingHoles());
+
+  d_preprocContext->spendResource(options::preprocessStep());
+
+
+  // Convert the original formula into a boolean formula and solve it
+  // with cryptominisat
+  SatSolver* d_satSolver = SatSolverFactory::createCryptoMinisat(smtStatisticsRegistry(), "non_clausal_simp_solver");
```

Comment:
Since `d_satSolver` is local here, you don't have to prefix it with `d_`. Variables should be prefixed with `d_` if they are data members of classes (https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#naming-1).

Further, use a `std::unique_ptr` so that you don't have to explicitly call delete.